mgmt/rest/plugin.go

```diff
@@ -259,7 +259,19 @@ func (s *Server) unloadPlugin(w http.ResponseWriter, r *http.Request, p httprout
 	respond(200, pr, w)
 }
 
-func (s *Server) getPlugins(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+func (s *Server) getPlugins(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	getPlugins(s, w, r, params)
+}
+
+func (s *Server) getPluginsByType(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	getPlugins(s, w, r, params)
+}
+
+func (s *Server) getPluginsByName(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	getPlugins(s, w, r, params)
+}
+
+func getPlugins(s *Server, w http.ResponseWriter, r *http.Request, params httprouter.Params) {
 	var detail bool
```

Comment:
Instead of having these redirect to the same place would it make sense to change [these](https://github.com/intelsdi-x/snap/blob/master/mgmt/rest/server.go#L286-L288) to all point to getPlugins() directly?